websites/E/Emby/presence.ts

```diff
@@ -408,59 +433,52 @@ function getUserId(): string {
 			localStorage.getItem("servercredentials3")
 		).Servers;
 
-		// server id available on browser location
-		if (location.hash.indexOf("?") > 0) {
-			for (const param of location.hash.split("?")[1].split("&")) {
-				if (param.startsWith("serverId")) {
-					for (const server of servers)
-						if (server.Id === param.split("=")[0]) return server.UserId;
-				}
-			}
-		} else return servers[0].UserId;
+		return (
+			servers.length === 1
+				? servers[0]
+				: servers.find(
+						(s: { Id: string }) =>
+							s.Id ===
+							new URLSearchParams(location.hash.split("?")[1]).get("serverId")
+				  )
+		).UserId;
 	}
 }
 
 // cache the requested media
-const media: Record<string, string | MediaInfo> = {};
+const mediaInfoCache = new Map<string, MediaInfo>();
 
 /**
  * obtainMediaInfo - obtain the metadata of the given id
  *
  * @param  {string} itemId id of the item to get metadata of
  * @return {object}        metadata of the item
  */
-async function obtainMediaInfo(itemId: string): Promise<string | MediaInfo> {
-	const pending = "pending";
-
-	if (media[itemId]) {
-		if (media[itemId] !== pending) return media[itemId];
+async function obtainMediaInfo(itemId: string): Promise<MediaInfo> {
+	if (mediaInfoCache.has(itemId)) return mediaInfoCache.get(itemId);
 
-		return;
-	}
+	let accessToken = ApiClient._serverInfo.AccessToken;
```

Comment:
You can probably use object destructuring here

Reply:
Fixed this one and the other at line 466